### ISSUE TITLE
Console formatting injection with --ansi and --diff on code with "<error>....</error>" etc.

### ISSUE THREAD

**Issue by ktomk** [FriendsOfPHP/PHP-CS-Fixer#1895]
When I run php-cs-fixer over code that contains symfony console component typical formatting strings and I output with the --ansi optionand --diff (in my case to review with `less -R`) it looks like that the source-code displayed in the diff output is getting replaced with colors when it contains color formatting codes.

~~~
$ php-cs-fixer --version
PHP CS Fixer version 1.11.2 by Fabien Potencier (41f7015)
~~~

Example command-line:
~~~
$ php-cs-fixer fix --config-file build/.php_cs --dry-run --diff --ansi | less -R
~~~

Exemplary output:

![selection_013](https://cloud.githubusercontent.com/assets/352517/15330706/4b9701be-1c5e-11e6-8d05-cc19d4836876.png)

Strings appear as empty, but the concrete code is:

~~~
    /**
     * @param OutputInterface $output
     * @param string $name
     * @param string $value
     * @param Constraints\Collection $constraints The constraint(s) to validate against.
     *
     * @return string
     */
    private function validateArgument(OutputInterface $output, $name, $value, $constraints)
    {
        if (strlen($value)) {
            $errors = $this->validateValue($name, $value, $constraints);
            if ($errors->count() > 0) {
                $output->writeln('<error>' . $errors[0]->getMessage() . '</error>');
            } else {
                return $value;
            }
        }

        $question = '<question>' . ucfirst($name) . ': </question>';

         ...
~~~

**Comment by SpacePossum** [comment]
The fixer doesn't pick up the `--ansi` flag itself I think, so if this is not a default flag on the SF console we might/could/should add it. 
Besides the flag we should always not replace the output with the colors on the `--diff` output itself.
Thanks for reporting I hope to look into this soon.

**Comment by ktomk** [comment]
Well the diff output has some nice colourings that look legit (sections headers, pluses and minuses), it was just these places highlighted in the screen-shot I was wondering about the colours. It took me a moment to understand that they mark no changes but were caused by an injection. For me this looks like a data-encoding problem so I reported it.

**Comment by keradus** [comment]
1. we dont have `--ansi` flag
2. isn't it because of that `less` call ?

**Comment by SpacePossum** [comment]
The `--ansi` is supported by default through the console package of SF.
Please see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1896

**Comment by ktomk** [comment]
@keradus: 1.) - like @SpacePossum wrote; 2.) No, `less` is not related.

**Comment by keradus** [comment]
Yeah, I think that @SpacePossum PR that was open after my question is enough answer

**Comment by keradus** [comment]
Closed by #1896
